Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Order.lean

```diff
@@ -6,28 +6,94 @@ Authors: Frédéric Dupuis
+import Mathlib.Analysis.NormedSpace.Star.Unitization
 
 /-! # Facts about star-ordered rings that depend on the continuous functional calculus
 
 This file contains various basic facts about star-ordered rings (i.e. mainly C⋆-algebras)
 that depend on the continuous functional calculus.
 
+We also put an order instance on `Unitization ℂ A` when `A` is a C⋆-algebra via
+the spectral order.
+
 ## Main theorems
 
 * `IsSelfAdjoint.le_algebraMap_norm_self` and `IsSelfAdjoint.le_algebraMap_norm_self`,
   which respectively show that `a ≤ algebraMap ℝ A ‖a‖` and `-(algebraMap ℝ A ‖a‖) ≤ a` in a
   C⋆-algebra.
 * `mul_star_le_algebraMap_norm_sq` and `star_mul_le_algebraMap_norm_sq`, which give similar
   statements for `a * star a` and `star a * a`.
+* `CstarRing.norm_le_norm_of_nonneg_of_le`: in a non-unital C⋆-algebra, if `0 ≤ a ≤ b`, then
+  `‖a‖ ≤ ‖b‖`.
+* `CstarRing.conjugate_le_norm_smul`: in a non-unital C⋆-algebra, we have that
+  `star a * b * a ≤ ‖b‖ • (star a * a)` (and a primed version for the `a * b * star a` case).
 
 ## Tags
 
 continuous functional calculus, normal, selfadjoint
 -/
 
+open scoped NNReal
+
+namespace Unitization
+
+variable {A : Type*} [NonUnitalNormedRing A] [CompleteSpace A] [Nontrivial A]
+  [PartialOrder A] [StarRing A] [StarOrderedRing A] [CstarRing A] [NormedSpace ℂ A] [StarModule ℂ A]
+  [SMulCommClass ℂ A A] [IsScalarTower ℂ A A]
+
+instance instPartialOrder : PartialOrder (Unitization ℂ A) := CstarRing.spectralOrder _
+
+instance instStarOrderedRing : StarOrderedRing (Unitization ℂ A) := CstarRing.spectralOrderedRing _
+
+lemma inr_le_iff (a b : A) (ha : IsSelfAdjoint a := by cfc_tac)
+    (hb : IsSelfAdjoint b := by cfc_tac) :
+    (a : Unitization ℂ A) ≤ (b : Unitization ℂ A) ↔ a ≤ b := by
+  -- TODO: prove the more general result for star monomorphisms and use it here.
+  rw [← sub_nonneg, ← sub_nonneg (a := b), StarOrderedRing.nonneg_iff_spectrum_nonneg (R := ℝ) _,
+    ← inr_sub ℂ b a, ← Unitization.quasispectrum_eq_spectrum_inr' ℝ ℂ]
+  exact StarOrderedRing.nonneg_iff_quasispectrum_nonneg _ |>.symm
+
+@[simp, norm_cast]
+lemma inr_nonneg_iff {a : A} : 0 ≤ (a : Unitization ℂ A) ↔ 0 ≤ a := by
+  by_cases ha : IsSelfAdjoint a
+  · exact inr_zero ℂ (A := A) ▸ inr_le_iff 0 a
+  · refine ⟨?_, ?_⟩
+    all_goals refine fun h ↦ (ha ?_).elim
+    · exact isSelfAdjoint_inr (R := ℂ) |>.mp <| .of_nonneg h
+    · exact .of_nonneg h
+
+end Unitization
+
+section generic
+
+variable {A : Type*} [TopologicalSpace A] [Ring A] [StarRing A] [PartialOrder A]
+  [Algebra ℂ A] [ContinuousFunctionalCalculus ℂ (IsStarNormal : A → Prop)]
+  [NonnegSpectrumClass ℝ A]
+
+lemma IsSelfAdjoint.coe_mem_spectrum_complex {a : A} {x : ℝ} (ha : IsSelfAdjoint a := by cfc_tac) :
+    (x : ℂ) ∈ spectrum ℂ a ↔ x ∈ spectrum ℝ a := by
+  simp [← ha.spectrumRestricts.algebraMap_image]
+
+lemma coe_mem_spectrum_real_of_nonneg {a : A} {x : ℝ≥0} (ha : 0 ≤ a := by cfc_tac) :
+    (x : ℝ) ∈ spectrum ℝ a ↔ x ∈ spectrum ℝ≥0 a := by
```

Comment:
You should be able to get away with just `Algebra ℝ A` here instead of `Algebra ℂ A`.